src/nvim/fileio.c

```diff
@@ -2771,6 +2771,9 @@ int buf_write(buf_T *buf, char *fname, char *sfname, linenr_T start, linenr_T en
               != 0) {
             SET_ERRMSG(_("E506: Can't write to backup file "
                          "(add ! to override)"));
+            XFREE_CLEAR(backup);
+            backup = NULL;
+            continue;
```

Comment:
We were missing the `continue` here on failure, so we were only every trying to create the backup file in the first directory (because we `break` out of the loop over all the backupdir options on line 2788 without checking `dirp` & `backup`). Before #9709, that break only happened _after_ we [checked that the creation of the backup worked](https://github.com/neovim/neovim/pull/9709/files#diff-1b7a1b0f3dbb1d3f4c6520ab4207aeedbbf7e8cfa129d7c710a64e06db20ae9dL2878-L2886)